src/engine/database/neo4j.rs

```diff
@@ -631,17 +638,30 @@ impl Transaction for Neo4jTransaction<'_> {
 
         let param_var = "param".to_string() + &sg.suffix();
         if !props.is_empty() {
+            let mut value_props : HashMap<String, Value> = HashMap::new();
             props.keys().enumerate().for_each(|(i, k)| {
```

Comment:
Similar to the above, I'm guessing something like

```
props.into_iter().enumerate().for_each(|(i, (k, v))| ...)
```

would allow you to take ownership and delete the line assigning to c, below.